src_python/fl_client.py

```diff
@@ -0,0 +1,223 @@
+                
+            else:
+                
+                self.rounds += 1
+                logging.info('_____________________________________________________ Training Round %s ____________________________________________________________',self.rounds)
+                logging.info('Global model v%s fetched',self.rounds - 1)
+
+                eval = self.MODEL.evaluate()
+
+                try:
+                    f1_train = (2 * eval[0][2] * eval[0][4]) / (eval[0][2] + eval[0][4])
+                    f1_test = (2 * eval[1][2] * eval[1][4]) / (eval[1][2] + eval[1][4])
+                except ZeroDivisionError as e:
+                    f1_train = "undefined"
+                    f1_test = "undefined"
+
+                logging.info('Global model v%s - Training set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds - 1, eval[0][0], eval[0][1],eval[0][2],eval[0][3],f1_train,eval[0][4])
+                logging.info('Global model v%s - Testing set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds - 1,  eval[1][0], eval[1][1],eval[1][2],eval[1][3],f1_test,eval[1][4])
+
+                
+                logging.info('Training started')
+                self.train()
+                logging.info('Training done')
+
+                # eval = self.MODEL.evaluate()
+
+                # f1_train = (2 * eval[0][2] * eval[0][4]) / (eval[0][2] + eval[0][4])
+                # f1_test = (2 * eval[1][2] * eval[1][4]) / (eval[1][2] + eval[1][4])
+                # logging.info('After Round %s - Local model - Training set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds, eval[0][0], eval[0][1],eval[0][2],eval[0][3],f1_train,eval[0][4])
+                # logging.info('After Round %s - Local model - Testing set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds, eval[1][0], eval[1][1],eval[1][2],eval[1][3],f1_test,eval[1][4])
+
+                logging.info('Sent local model to the server')
+                self.send_model()
+
+
+if __name__ == "__main__":
+
+    from models.supervised import Model
+
+    if 'IP' not in args.keys()  or args['IP'] == 'localhost':
+        args['IP'] = socket.gethostname()
+
+    if 'PORT' not in args.keys():
+        args['PORT'] = 5000
+
+    if 'epochs' not in args.keys():
+        args['epoch'] = 10
+
+    logging.warning('####################################### New Training Session #######################################')
+    logging.info('Client started, graph ID %s, partition ID %s, epochs %s',args['graph_id'],args['partition_id'],args['epochs'])
+
+    path_nodes = args['path_nodes'] + args['graph_id'] + '_nodes_' + args['partition_id'] + ".csv"
```

Comment:
Need to keep these files also inside the JasmineGraph directory structure starting from /var/tmp/jasminegraph

Reply:
This absolute path is provided when the scripts are called

Reply:
Based on the discussion we had the `path_nodes` variable contains the absolute path for the CSV file. That means the variable `path_nodes` starts with `/var/tmp/jasminegraph`.